Bottom: 99b70675ea68c5f72ba03a31cab6e6df26ce501d
Top:    536c6b1a10f8ae6feeb68431088ba2a92b11df12
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-28 03:55:56 +0000

Refresh of debug-wonky-copper-polygon

---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 2f8a8cb..6db76b1 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -23,6 +23,7 @@
 #include "misc.h"
 #include "hid/hidint.h"
 
+
 //#define REVERSED_PCB_CONTOURS 1 /* PCB Contours are reversed from the expected CCW for outer ordering - once the Y-coordinate flip is taken into account */
 #undef REVERSED_PCB_CONTOURS
 
@@ -1436,19 +1437,19 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
       {
         fprintf (stderr, "Accumulating elements from layer %i\n", GetLayerNumber (PCB->Data, layer));
 
-        r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
-        r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
-        r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
+//        r_search (layer->line_tree, &bounds, NULL, line_copper_callback, &info);
+//        r_search (layer->arc_tree,  &bounds, NULL, arc_copper_callback, &info);
+//        r_search (layer->text_tree, &bounds, NULL, text_copper_callback, &info);
         r_search (layer->polygon_tree, &bounds, NULL, polygon_copper_callback, &info);
       }
     END_LOOP;
 
-    fprintf (stderr, "Accumulating pin + via pads\n");
-    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+//    fprintf (stderr, "Accumulating pin + via pads\n");
+//    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+//    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 1
+#if 0
     if (group == top_group ||
         group == bottom_group)
       {
@@ -1502,6 +1503,7 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
                               NULL));
 
 
+    break;
   }
 
 
@@ -1512,7 +1514,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 //  PCB->Data->outline_valid = true;
 //  gui->invalidate_all ();
 
-  poly_Free (&info.poly);
+  /* ASSUME THERE IS A POLYGON WHERE WE KNOW WE PUT ONE... */
+  ((PolygonType *)PCB->Data->Layer[1].Polygon->data)->Clipped = info.poly;
+  gui->invalidate_all ();
+
+//  poly_Free (&info.poly);
 
   return objects;
 }
diff --git a/src/polygon1.c b/src/polygon1.c
index 5c09126..2372153 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -39,6 +39,7 @@
       are marked
 */
 
+#undef NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -109,13 +110,13 @@ int vect_inters2 (Vector A, Vector B, Vector C, Vector D, Vector S1,
 
 #define error(code)  longjmp(*(e), code)
 
-#undef DEBUG_INTERSECT
-#undef DEBUG_LABEL
-#undef DEBUG_ALL_LABELS
-#undef DEBUG_JUMP
-#undef DEBUG_GATHER
-#undef DEBUG_ANGLE
-#undef DEBUG
+#define DEBUG_INTERSECT
+#define DEBUG_LABEL
+#define DEBUG_ALL_LABELS
+#define DEBUG_JUMP
+#define DEBUG_GATHER
+#define DEBUG_ANGLE
+#define DEBUG
 #ifdef DEBUG
 #define DEBUGP(...) pcb_fprintf(stderr, ## __VA_ARGS__)
 #else
